src/compile/mark/encode/defined.ts

```diff
@@ -1,49 +1,48 @@
-import {POSITION_SCALE_CHANNELS} from '../../../channel';
-import {ScaleChannel} from '../../../channel';
+import {isCountingAggregateOp} from '../../../aggregate';
+import {isScaleChannel} from '../../../channel';
 import {Value} from '../../../channeldef';
-import {hasContinuousDomain} from '../../../scale';
-import {Dict, keys} from '../../../util';
+import {fieldValidPredicate} from '../../../predicate';
 import {VgEncodeEntry} from '../../../vega.schema';
-import {getMarkPropOrConfig, signalOrValueRef} from '../../common';
+import {signalOrValueRef} from '../../common';
+import {getScaleInvalidDataMode, shouldBreakPath} from '../../invalid/ScaleInvalidDataMode';
 import {UnitModel} from '../../unit';
-import {fieldInvalidPredicate} from './valueref';
 
+/**
+ * Create Vega's "defined" encoding to break paths in a path mark for invalid values.
+ */
 export function defined(model: UnitModel): VgEncodeEntry {
   const {config, markDef} = model;
 
-  const invalid = getMarkPropOrConfig('invalid', markDef, config);
-  if (invalid) {
-    const signal = allFieldsInvalidPredicate(model, {channels: POSITION_SCALE_CHANNELS});
+  // For each channel (x/y), add fields to break path to a set first.
+  const fieldsToBreakPath = new Set<string>();
 
-    if (signal) {
-      return {defined: {signal}};
+  model.forEachFieldDef((fieldDef, channel) => {
+    let scaleType;
+    if (!isScaleChannel(channel) || !(scaleType = model.getScaleType(channel))) {
```

Comment:
nit: I prefer not to use assignment in condition in production code.

Reply:
Not sure why? I don't think it hurts readability that much in this case. It actually compact the code. 

In any case, Vega-Lite uses this quite regularly, so let's not waste time with this.

Reply:
https://eslint.org/docs/latest/rules/no-cond-assign Ppl might easily mess the code between `foo = bar` and `foo == bar`. You can keep it if it's an existing standard.